### PR TITLE
clear voter claims when resetting demo

### PR DESCRIPTION
**Pull request by liewegas**
This obscures a complete history in the database, but these are demo
voters--we really don't care.  (And the table doesn't provide a way
to ignore claims pre-reset.)

### CODE REVIEW

**Comment by liewegas** [comment]
Tryn ran into this one last night!

**Review by tomerovadia — COMMENTED** [review]
Is the purpose of this change ease of developing/QA, or something else?

The original use case for demo conversations requires that this `volunteer_voter_claims` history be persisted (for volunteer-to-volunteer practice and admin review of volunteer readiness). The history of voter claims provides a way to determine which volunteers answered which demo conversations, to be matched against their messages and then evaluated to determine if volunteers are prepared to take on real voters.

We can add an `archived` column similarly to what exists for the `messages` table: https://github.com/tomerovadia/VoterHelpLine/pull/79

That also allows us to keep the `volunteer_voter_claims` table write-only.

**Comment by liewegas** [comment]
Much better.  Updated!

File: src/db_api_util.ts [hunk 1]
```diff
@@ -531,6 +531,33 @@ export async function archiveMessagesForDemoVoter(
   }
 }
 
+export async function archiveDemoVoterClaims(
```
Comment (tomerovadia) [inline src/db_api_util.ts]: nit: can we name this archiveDemoVolunteerVoterClaims?

File: src/db_api_util.ts [hunk 2]
```diff
@@ -531,6 +531,33 @@ export async function archiveMessagesForDemoVoter(
...
+  userId: string,
+  twilioPhoneNumber: string
+): Promise<void> {
+  logger.info(`ENTERING DBAPIUTIL.clearDemoVoterClaims`);
```
Comment (tomerovadia) [inline src/db_api_util.ts]: update log line

File: src/db_api_util.ts [hunk 3]
```diff
@@ -531,6 +531,33 @@ export async function archiveMessagesForDemoVoter(
...
+      SET archived = true
+      WHERE
+        is_demo = true
+        AND user_id = $1
+        AND twilio_phone_number = $2`,
+      [userId, twilioPhoneNumber]
+    );
+
+    logger.info(
+      `DBAPIUTIL.clearDemoVoterClaims: Successfully cleared demo voter claims.`
```
Comment (tomerovadia) [inline src/db_api_util.ts]: if you update the function name let's update the log line here too

File: sql/20201022.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE volunteer_voter_claims
```
Comment (tomerovadia) [inline sql/20201022.sql]: can you update the name of this file to today's date before merging?

**Review by tomerovadia — CHANGES_REQUESTED** [review]
Looks good!

**Review by tomerovadia — APPROVED** [review]

**Pull request merged**